src/hooks/useFetchPriceHistories/useFetchPriceHistories.ts

```diff
@@ -16,47 +16,53 @@ export const useFetchPriceHistories = (assetIds: AssetId[], timeframe: HistoryTi
   const symbol = useAppSelector(selectSelectedCurrency)
 
   useEffect(() => {
-    dispatch(findPriceHistoryByAssetIds.initiate({ assetIds, timeframe }))
+    assetIds.forEach(assetId => {
+      dispatch(marketApi.endpoints.findPriceHistoryByAssetId.initiate({ assetId, timeframe }))
+    })
```

Comment:
q: I don't know if it's managed under the hood but by reading this little piece of code without knowing the underlying context, I feel like we are rerunning all the price history fetching for every assets if a small piece inside `assetIds` changes

Reply:
We don't `forceRefetch` here so we we won't refetch, only fetch fresh ones here